aptos-move/aptos-vm-types/src/module_write_set.rs

```diff
@@ -57,12 +60,21 @@ impl ModuleWriteSet {
     pub fn write_op_info_iter_mut<'a>(
         &'a mut self,
         executor_view: &'a dyn ExecutorView,
+        module_storage: &'a impl AptosModuleStorage,
+        is_loader_v2_enabled: bool,
     ) -> impl Iterator<Item = PartialVMResult<WriteOpInfo>> {
-        self.write_ops.iter_mut().map(|(key, op)| {
+        self.write_ops.iter_mut().map(move |(key, op)| {
+            let prev_size = if is_loader_v2_enabled {
+                module_storage
+                    .fetch_module_size_by_state_key(key)?
+                    .unwrap_or(0) as u64
+            } else {
+                executor_view.get_module_state_value_size(key)?.unwrap_or(0)
```

Comment:
Can you elaborate in which case the two sizes would be different?

Reply:
As mentioned in the sync: sizes are the same, we just need to fetch them through different APIs